resources/perf.webkit.org/public/v3/pages/page-router.js

```diff
@@ -97,8 +98,12 @@ class PageRouter {
 
     _hashDidChange()
     {
-        if (unescape(location.hash) == this._hash)
+        let unescapedHash = unescape(location.hash);
+        if (unescapedHash == this._hash)
             return;
+        if(unescapedHash == this._previousHash)
+            return;
+        this._previousHash = unescapedHash;
         this.route();
         this._hash = null;
```

Comment:
it's not clear to me why `this._hash` is set to null here. In `_updateURLState` above it looks like we want to keep it synced with location.hash.
Then if we don't reset it, we probably don't need _previousHash and the check above would work, right?

Reply:
That's a good point, much simpler to do that. Fixed :)